streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -97,31 +96,27 @@ public void commit() {
     }
 
     @Override
-    public void close() {
-        //no-op
+    public void suspend() {
+        commit();
     }
 
     @Override
-    public void initTopology() {
-        //no-op
+    public void close() {
+        try {
+            commit();
+            closeStateManager(true);
+        } catch (final RuntimeException e) {
```

Comment:
EDIT: just realizing that we are re-throwing the exception anyways after re-closing the state managers. So this should be fine.